src/types.rs

```diff
@@ -240,6 +240,60 @@ impl BlockTime {
     }
 }
 
+/// Balance differentiated in various categories
+#[derive(Serialize, Deserialize, Debug, PartialEq, Clone, Default)]
+pub struct Balance {
+    /// All coinbase outputs not yet matured
+    pub immature: u64,
+    /// Unconfirmed UTXO generated by a wallet tx
+    pub trusted_pending: u64,
+    /// Unconfirmed UTXO received from an external wallet
+    pub untrusted_pending: u64,
+    /// All rest available balance
```

Comment:
Maybe this is more clear if we document it as "Confirmed and immediately spendable balance"

Reply:
...which actually kind of opens a can of worms re: timelocked UTXOs.

In some cases it's pretty obvious what's timelocked and what's not, for example if your script is a simple `and(pk(key),after(value))`.

However, sometimes your timelock is only part of a branch, which means that your UTXOs may or may not be locked depending on how you'd like to spend them. I'm not sure what's the best way to fix this, ideally this function would take some kind of parameter to specify this, so that we can give an accurate answer to the user.

That said, I don't think it's a good idea to hold this PR while we discuss it, maybe we can open a different issue to keep track of that.

Reply:
I've opened #676 to keep track of this, resolving this convo for now